Remotion/Web/Core/UI/SmartPage.cs

```diff
@@ -525,6 +525,11 @@ protected override void OnInit (EventArgs e)
   {
     base.OnInit(e);
     RegisterRequiresControlState(this);
+
+    if (!IsPostBack)
+      ClientScript.RegisterHiddenField(this, "SmartPage_InstanceToken", Guid.NewGuid().ToString());
+    else
+      ClientScript.RegisterHiddenField(this, "SmartPage_InstanceToken", Request.Params["SmartPage_InstanceToken"]);
```

Comment:
For consistent implementation pattern:
See how SmartPageInfo handles reading the c_smartScrollingID during prerender.